In [2]:
# Importing Libraries
import pandas as pd
import numpy as np
import os
import matplotlib.pyplot as plt
import seaborn as sns
import scipy

In [4]:
# Generating a path
path = '/Users/samarjitgehdu/Documents/Python/Instacart Basket Analysis'

In [6]:
# Importing the Dataframe
df = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'customers.csv'))

In [8]:
# Data Wrangling
df.head()

user_id First Name    Surnam  Gender       STATE  Age date_joined  \
0    26711    Deborah  Esquivel  Female    Missouri   48    1/1/2017   
1    33890   Patricia      Hart  Female  New Mexico   36    1/1/2017   
2    65803    Kenneth    Farley    Male       Idaho   35    1/1/2017   
3   125935   Michelle     Hicks  Female        Iowa   40    1/1/2017   
4   130797        Ann   Gilmore  Female    Maryland   26    1/1/2017   

   n_dependants fam_status  income  
0             3    married  165665  
1             0     single   59285  
2             2    married   99568  
3             0     single   42049  
4             1    married   40374

In [10]:
df.shape

(206209, 10)

In [12]:
# Renaming Columns Names
df = df.rename(
    columns = {
        'Surnam' : 'Last Name',
        'STATE' : 'State',
        'date_joined' : 'Date Joined',
        'n_dependants' : 'Number of Dependents',
        'fam_status' : 'Family Status',
        'income' : 'Income'
    }
)

In [14]:
df.head()

user_id First Name Last Name  Gender       State  Age Date Joined  \
0    26711    Deborah  Esquivel  Female    Missouri   48    1/1/2017   
1    33890   Patricia      Hart  Female  New Mexico   36    1/1/2017   
2    65803    Kenneth    Farley    Male       Idaho   35    1/1/2017   
3   125935   Michelle     Hicks  Female        Iowa   40    1/1/2017   
4   130797        Ann   Gilmore  Female    Maryland   26    1/1/2017   

   Number of Dependents Family Status  Income  
0                     3       married  165665  
1                     0        single   59285  
2                     2       married   99568  
3                     0        single   42049  
4                     1       married   40374

In [16]:
df.describe()

user_id            Age  Number of Dependents         Income
count  206209.000000  206209.000000         206209.000000  206209.000000
mean   103105.000000      49.501646              1.499823   94632.852548
std     59527.555167      18.480962              1.118433   42473.786988
min         1.000000      18.000000              0.000000   25903.000000
25%     51553.000000      33.000000              0.000000   59874.000000
50%    103105.000000      49.000000              1.000000   93547.000000
75%    154657.000000      66.000000              3.000000  124244.000000
max    206209.000000      81.000000              3.000000  593901.000000

In [18]:
# Finding missing values
Missing_Values = df.isnull().sum()

In [20]:
Missing_Values

user_id                     0
First Name              11259
Last Name                   0
Gender                      0
State                       0
Age                         0
Date Joined                 0
Number of Dependents        0
Family Status               0
Income                      0
dtype: int64

In [22]:
# Finding Duplicates Rows
Duplicates = df.duplicated()

In [24]:
# Sum of the total duplicates 
Num_of_Duplicates = Duplicates.sum()

In [26]:
Num_of_Duplicates

0

In [28]:
# Verifying column datatypes
df.dtypes

user_id                  int64
First Name              object
Last Name               object
Gender                  object
State                   object
Age                      int64
Date Joined             object
Number of Dependents     int64
Family Status           object
Income                   int64
dtype: object

### Observation
Since no duplicates are found and the datatypes appear to be in their correct format. Other than the missing values of 11259 for first name which won't affect the main objective of the Analysis.

### Combining Dataframe

In [34]:
# Importing Dataframe
ords_prods_merge = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'aggregated_ords_prods_merge.pkl'))

In [36]:
# Combining Dataframe customer and ords_prods_merge
Merged_df = pd.merge(ords_prods_merge, df, on = 'user_id', how = 'inner')

In [38]:
# Verify the merged dataframe
Merged_df.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2539329        1             1                   2                  8   
2   2539329        1             1                   2                  8   
3   2539329        1             1                   2                  8   
4   2539329        1             1                   2                  8   

   days_since_prior_order  product_id  add_to_cart_order  reordered _merge  \
0                     NaN         196                  1          0   both   
1                     NaN       14084                  2          0   both   
2                     NaN       12427                  3          0   both   
3                     NaN       26088                  4          0   both   
4                     NaN       26405                  5          0   both   

   ...         frequency_flag  First Name  Last Name  Gender    State Age  \
0  ...  Non_frequent_customer       Linda     Nguyen  Female  Alabama  31   
1  ...  Non_frequent_customer       Linda     Nguyen  Female  Alabama  31   
2  ...  Non_frequent_customer       Linda     Nguyen  Female  Alabama  31   
3  ...  Non_frequent_customer       Linda     Nguyen  Female  Alabama  31   
4  ...  Non_frequent_customer       Linda     Nguyen  Female  Alabama  31   

  Date Joined Number of Dependents Family Status  Income  
0   2/17/2019                    3       married   40423  
1   2/17/2019                    3       married   40423  
2   2/17/2019                    3       married   40423  
3   2/17/2019                    3       married   40423  
4   2/17/2019                    3       married   40423  

[5 rows x 34 columns]

In [40]:
Merged_df.shape

(32404859, 34)

### Exporting the file in pickle format

In [43]:
Merged_df.to_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'ords_prods_cust_merge.pkl'))